components/collective-page/hero/Hero.js

```diff
@@ -242,25 +244,25 @@ const Hero = ({ collective, host, isAdmin, onPrimaryColorChange }) => {
                       values={{
                         FiscalHost: <DefinedTerm term={Terms.FISCAL_HOST} color="black.700" />,
                         hostName: (
-                          <LinkCollective collective={host}>
-                            <Span data-cy="fiscalHostName" color="black.700">
-                              {host.name}
+                          <LinkCollective collective={host} data-cy="fiscalHostName" noTitle>
+                            <Span color="black.700">
+                              <TruncatedTextWithTooltip value={host.name} />
                             </Span>
                           </LinkCollective>
                         ),
                       }}
                     />
                   </Container>
-                  {collective.connectedTo.length !== 0 && (
+                  {!parentIsHost && collective.connectedTo?.length > 0 && (
```

Comment:
Doesn't seem necessary since we already check it directly above

```suggestion
                  {!parentIsHost && collective.connectedTo.length > 0 && (
```